### PR TITLE
ch4: code cleanup

### PR DESCRIPTION
**Pull request by yfguo**
Running maint/code-cleanup.sh on all CH4 code.

### CODE REVIEW

**Comment by yfguo** [comment]
I am quite surprised that there are many files that need cleanup. It might be the appropriate moment to do that now. I would encourage everyone to do a cleanup before push a new PR.

**Comment by wesbland** [comment]
Are you sure you didn’t change the script here? We’d notice if these changes had been missed.

**Comment by hzhou** [comment]
There were two new versions of gnu indent in 2018. We probably should report and restrict the version.

File: src/include/mpir_attr.h [hunk 1]
```diff
@@ -39,11 +39,11 @@ MPII_Attr_copy_c_proxy(MPI_Comm_copy_attr_function * user_function,
                        MPIR_Attr_type attrib_type, void *attrib, void **attrib_copy, int *flag);
 
 typedef struct copy_function {
-    int (*C_CopyFunction) (int, int, void *, void *, void *, int *);
-    void (*F77_CopyFunction) (MPI_Fint *, MPI_Fint *, MPI_Fint *, MPI_Fint *,
-                              MPI_Fint *, MPI_Fint *, MPI_Fint *);
-    void (*F90_CopyFunction) (MPI_Fint *, MPI_Fint *, MPI_Aint *, MPI_Aint *,
-                              MPI_Aint *, MPI_Fint *, MPI_Fint *);
+    int (*C_CopyFunction)(int, int, void *, void *, void *, int *);
+    void (*F77_CopyFunction)(MPI_Fint *, MPI_Fint *, MPI_Fint *, MPI_Fint *,
```
Comment (hzhou) [inline src/include/mpir_attr.h]: This is where the new version gets right, +1, although I don't really care one way or the other as long as it's consistent.

File: src/include/mpir_dataloop.h [hunk 2]
```diff
@@ -414,36 +414,36 @@ void
 MPIR_Segment_manipulate(DLOOP_Segment * segp,
                         DLOOP_Offset first,
                         DLOOP_Offset * lastp,
-                        int (*piecefn) (DLOOP_Offset * blocks_p,
+                        int (*piecefn)(DLOOP_Offset * blocks_p,
```
Comment (hzhou) [inline src/include/mpir_dataloop.h]: Both versions misses `*` as pointer unary.

**Comment by hzhou** [comment]
Ubuntu 18.04 still uses gnu indent version `2.2.11`, and I suspect so are most distributions. Mac uses brew recipe (is that right?) so it picks up the later version. Since the update do not really bring significant benefits -- only inflates the git history, I would suggest keep the version `2.2.11` until major stable distribution upgrades. I do like to keep the version check commit -- but check against `2.2.11` instead -- it is more important to keep consistency rather than one way or the other. For developers using (latest) Mac, we always can manually install version `2.2.11`, shouldn't be a tall order for our developers :)

**Comment by pavanbalaji** [comment]
> I would suggest keep the version 2.2.11 until major stable distribution upgrades

+1

**Comment by wesbland** [comment]
Sounds good. If anyone has a Homebrew recipe for 2.2.11, let me know.

**Comment by hzhou** [comment]
@wesbland I just tried install directly using 2.2.11 tarball, tried simply `./configure && make`, and it complains on "missing texi2html", which is typical trying to build gnu packages on non-linux. I simply edited `Makefile`, search for `^SUBDIRS`, and deleted the `doc` target. I personally also deleted `info` and `man` as I use google anyway. :)

**Comment by wesbland** [comment]
Yeah, I’m just trying to get it to work with Homebrew so I can pin the package.

**Comment by hzhou** [comment]
I am too lazy (to really learn home-brew) so I often resort to hack when the default doesn't work :)

**Comment by wesbland** [comment]
I created a patch in a gist that lets you install 2.2.11 in case anyone wants it:

https://gist.github.com/wesbland/501063f151c1eb815d8001abf2285cbe

You need to be inside `/usr/local/Homebrew/Library/Taps/homebrew/homebrew-core` to apply it.

**Pull request closed without merging**